### DISCUSSION TITLE
Intonation from measured recorder is incorrect

### DISCUSSION THREAD

**Discussion by friedelwolff** [edwardkort/WWIDesigner#119] [Whistle Study] [answered]
Thanks for the software. I'm digging in trying to make a basic recorder model, and learning in the process. I'm running into difficulties, possibly due to making some incorrect assumptions.

I'm using a Native American Flute study. (I had less success with the Whistle study. Maybe due to where I'm from it isn't even clear exactly what kind of instrument "whistle" refers to.)

I have a simple instrument design with only a few notes and fingerings, but with holes from a working recorder I have here. The physical instrument has good intonation. Although my measurements are probably not 100% accurate, I believe it should be good enough to get me started.

The strange thing is that even with only 16 notes, the fipple factor is very low (less than 0.1 in most cases), while I understood it should be around 1.0. I understood it is dimensionless, so didn't think that me using millimetres should make a difference.

Secondly, even if my reference note is in tune with the calculated fipple factor, the intonation of the remaining 15 notes is bad (net error > 50, deviation > 120), with several notes two semitones out of tune. I assumed that since I start from known-good measurements for hole placement and dimensions (or something quite close to it), it should have pretty good intonation from the start.

Any idea what could be wrong in my model? I attach the instrument XML file below.

I'd like to get to a point where I can model the typical "split" holes at the bottom of a recorder, and the "half" thumb hole. Any pointers or advice would be very kind.

[altblokfluit.zip](https://github.com/edwardkort/WWIDesigner/files/5838477/altblokfluit.zip)

**Comment by bhp1** [comment]
Can you supply a tuning file to go with that, with fingerings and actual measured playing frequencies that go with the fingerings?  When you measure the playing frequencies, don't try to blow the notes in tune, just play them as they come, going up and down the "scale", as it were.

The Whistle Study should work with most fipple flutes, including your basic recorder.  WIDesigner may not do so well in general with the split holes or half-holing, but we can give it a try.

**Reply by friedelwolff** [reply]
Thank you for the reply. I attach the tuning file I used, but it was done with the Native American Flute study in mind, so does not contain playing frequencies.

[altblokfluit.zip](https://github.com/edwardkort/WWIDesigner/files/5840056/altblokfluit.zip)

I don't have a device, but I guess I can probably use some software for that. I'm still a bit confused about where the measured frequencies and the target frequencies are edited - the min/max columns I think I understand. Is all the N/A columns in the tuning calculation due to not having the min/max frequencies?

Would you say that the whistle study is the right tool for me to use (rather than the native american flute study)?

**Comment by bhp1** [comment, marked as answer]
Your tuning is in the wrong octave.  The bell note for that instrument would be F4, around 349 Hz, not F3.  Double all the frequencies in your tuning file, and the whistle study will be able to predict frequencies for that instrument.

To calibrate the model, you'll need a tuner that can measure and report frequencies in Hertz.  On a computer with a microphone, you can use the shakuhachi tuner at https://shaku6.com/s8tuner.php.  For an Android smartphone, there are a number of tuner apps; I use one called, simply, ["Instrument Tuner"](https://play.google.com/store/apps/details?id=instrument.tuner).  I'm sure there are similar apps for iPhones.  The [instructions for calibrating the whistle model are here](https://github.com/edwardkort/WWIDesigner/wiki/Working-With-the-Whistle-Study-Model#calibrating-the-whistle-model).  While you look for a tuner, I will see about making those instructions clearer.

To run optimizations on that instrument, particularly with the split holes, you'll need to create [customized constraints](https://github.com/edwardkort/WWIDesigner/wiki/Optimizing-a-Whistle-or-Flute-Design#3b-optimize-the-design-with-custom-constraints).

**Reply by edwardkort** [reply]
Ah, the joys of European scientific notation. In Europe, middle C is C3, in the new world it is C4. And BTW, I quite often craft NAFs with the double bottom hole found in recorders. WIDesigner handles this with no problem; just make sure you enter the thinner wall that you are likely to craft around these holes.

I haven't tried to blueprint any of my recorders to see how well WIDesigner predicts the tuning using either the NAF or Whistle studies. It's on my list, but not very high.

Cheers,
Edward Kort
Wood Wind

**Reply by friedelwolff** [reply]
Sorry for my silly mistake about the octave!

@edwardkort, I'm not sure I understand what you mean with the "thinner wall". In the instrument I attached, you can see that the two wholes overlap (as visualised by WIDesigner). I was wondering if that would pose problems. But the dimensions are what I measured on the instruments. They are next to each other on the length axis of the instrument.

**Reply by edwardkort** [reply]
The hole height is the wall thickness at the hole.  In your instrument file, the wall thicknesses for holes 0 - 5 is about 4.5 mm; for holes 6a/6b slightly less; for holes 7a/7b almost twice as thick. All of my recorders have a channel at the doubled holes to aid in finger location and (primarily) to make the holes smaller due to the thinner wall at those holes. I would guess that the height at the doubled holes is 2 - 2.5 mm.

When estimating tuning, multiple holes at the same position are handled just fine. When I do an optimization with doubled holes, in the constraints I set the minimum hole spacing to something like 0.1 mm so that the optimizer doesn't try to swap the holes. But practically I don't know whether that is an issue.

**Reply by friedelwolff** [reply]
I measured 7a and 7b as best as I can! I was also surprised to see that they are that deep in the instrument body. They are in the part of the instrument that already starts to flare out towards the end. I'll try to review the measurements and report back.

I implemented an even simpler model without half holes on a much smaller recorder I have here starting at C6 and things work out and the tuning corresponds well with the instrument up to D7 but the next one (E♭) is badly calculated and doesn't correspond to the instrument. I fear that it is difficulty with the second register, or something I measured wrong. I'll review everything and get back. I'd like to contribute recorder templates for the next version to ease things for the next person like me.

**Reply by edwardkort** [reply]
There are two significant differences between the Native American Flute study and the Whistle study:
1. The underlying mouthpiece model
2. The strategies implemented for optimizing the design.
The native American flute (NAF) has typically a smaller aspect ratio (bore length/bore diameter) and a uniform bore diameter as compared to a recorder. As such, the breath pressure across its entire tonal range is fairly constant. The NAF mouthpiece model reflects this simplicity. On the other hand a recorder, with its significant bore taper, supports increased breath pressure when playing upper register and octave notes. Clean intonation when playing these notes requires that increased pressure.

So your results are not surprising. That is why I recommend using the Whistle study, which supports the calibration of this increased breath pressure, when designing recorders. But Burton, who is the principle developer of the Whistle study, may have a different opinion.

**Reply by bhp1** [reply]
The Whistle Study would definitely be my recommendation for recorders.  

It is reassuring that you've had success modelling the double holes in WIDesigner.
